_posts/news/2016-10-17-announcing-scalajs-0.6.13.md

```diff
@@ -0,0 +1,157 @@
+* PhantomJS: `jsEnv := PhantomJSEnv().value`.
+
+Remember that you can also use Selenium with Scala.js, using [scalajs-env-selenium](https://github.com/scala-js/scala-js-env-selenium).
+
+## Deprecation: `@JSGlobalScope` replaces `extends js.GlobalScope`
+
+As indicated by a deprecation warning, `extends js.GlobalScope` should not be used anymore.
+Instead, you should annotate the object with `@JSGlobalScope`.
+For example, this old snippet:
+
+{% highlight scala %}
+import scala.scalajs.js
+
+@js.native
+object Foo extends js.GlobalScope
+{% endhighlight %}
+
+should be replaced with
+
+{% highlight scala %}
+import scala.scalajs.js
+import js.annotation._
+
+@js.native
+@JSGlobalScope
+object Foo extends js.Object
+{% endhighlight %}
+
+## `@JSImport` and emitting CommonJS modules
+
+This is a long-awaited feature!
+Scala.js can now emit CommonJS modules (i.e., those used by Node.js, as well as several bundlers).
+
+### Enabling CommonJS module
+
+You can enable emission of a CommonJS module with the following sbt setting:
+
+{% highlight scala %}
+scalaJSModuleKind := ModuleKind.CommonJSModule
+{% endhighlight %}
+
+When emitting a CommonJS module, top-level `@JSExport`ed classes and objects are stored in the `exports` object, so that they can be required from other CommonJS modules.
+
+Obviously, this requires that you use Node.js for `sbt run` and `sbt test` (should you use them at all), and that you use a JavaScript bundler such as [Webpack](https://webpack.github.io/docs/) to create a bundle fit for use in the browser.
+At the moment, Scala.js does not provide any facility to do so.
+
+Emitting CommonJS modules is also not compatible with `persistLauncher := true`, as a different launcher needs to be emitted for fastOpt versus fullOpt.
+
+More importantly, the CommonJS module support is completely incompatible with the `jsDependencies` mechanism.
+If you emit CommonJS modules, you have to manage your JavaScript dependencies on your own (possibly through `npm`).
+
+### Importing stuff from other CommonJS modules
+
+To import other CommonJS modules from Scala.js, you should use `@JSImport` ([Scaladoc]({{ site.production_url }}/api/scalajs-library/0.6.13/#scala.scalajs.js.annotation.JSImport)).
+Semantically speaking, `@JSImport` is an [ECMAScript 2015 import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import), but Scala.js desugars it into a CommonJS `require`.
+Let us see an example first:
+
+{% highlight scala %}
+import scala.scalajs.js
+import js.annotation._
+
+@js.native
+@JSImport("bar.js", "Foo") // JS: import Foo from "bar.js"
+class Foobaz(var bar: Int) extends js.Object
+
+val foo = new Foobaz(5) // JS: new Foo(5)
+{% endhighlight %}
+
+If the module exports top-level functions or variables, you should create an `object` representing the module itself, like this:
+
+{% highlight scala %}
+@js.native
+@JSImport("bar.js", JSImport.Namespace) // JS: import * as bar from "bar.js"
```

Comment:
Same here:

```scala
// ES5: var bar = require("bar.js")
// ES6: import * as bar from "bar.js"
```

Especially here, most readers can probably much more relate to `var x = require("x.js");` than the ES2015 equivalent.